### PR TITLE
Feat: implement Deserializer for BlockId and EventFilter

### PR DESCRIPTION
**Pull request by bigherc18**

### CODE REVIEW

**Comment by bigherc18** [comment]
@xJonathanLEI All checks pass for this PR, can you please take a look at it ?

File: starknet-providers/src/jsonrpc/models/codegen.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ pub struct ResultPageRequest {
 ///
 /// Event information decorated with metadata on where it was emitted.
 #[serde_as]
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
```
Comment (xJonathanLEI) [inline starknet-providers/src/jsonrpc/models/codegen.rs:41]: As noted on the first line of this file, this file is automatically generated and thus should not be manually edited, as any changes will be overwritten subsequently.

File: starknet-providers/tests/models.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+use serde::Deserialize;
```
Comment (xJonathanLEI) [inline starknet-providers/tests/models.rs:1]: The `tests` folder is for [integration tests](https://doc.rust-lang.org/book/ch11-03-test-organization.html#integration-tests). You might want to place these tests elsewhere.

File: starknet-providers/src/jsonrpc/models/mod.rs [hunk 3]
```diff
@@ -224,6 +228,58 @@ impl Serialize for BlockId {
...
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter.write_str("a block_hash, block_number, latest or pending")
+    }
+
+    fn visit_map<A>(self, mut map: A) -> Result<Self::Value, A::Error>
+    where
+        A: serde::de::MapAccess<'de>,
+    {
+        match map.next_key::<String>()?.unwrap().as_str() {
```
Comment (xJonathanLEI) [inline starknet-providers/src/jsonrpc/models/mod.rs:243]: I think it's not safe to unwrap here?

**Review by xJonathanLEI — COMMENTED** [review]
Thanks for the PR but I think this is not quite ready to merge yet.

**Comment by xJonathanLEI** [comment]
Superseded by #350.

**Pull request closed without merging**